lib/network.js

```diff
@@ -13,7 +13,13 @@ module.exports = function (archive, opts, cb) {
     stream: opts.stream
   }, opts)
   var swarm = disc(swarmDefaults(swarmOpts))
-  swarm.once('error', function () {
+  swarm.once('error', function (err) {
+    // Don't magically use any port if the users has requested a specific port
+    swarm.emit('errorport', err)
```

Comment:
Can we use `emit('error', new Error('specified port not available'))` (or similar) for use cases outside the dat cli. Will require less documentation than a custom event =).

(Not sure if I left this before but not seeing it. Apologies if its duplicate.)